### PR TITLE
implemented AvailableServices and Service in serviceManager

### PR DESCRIPTION
**Pull request by ineiti**
Implements #565

### CODE REVIEW

**Comment by nikkolasg** [comment]
OK for me.

File: sda/service.go [hunk 1]
```diff
@@ -272,19 +272,21 @@ func (s *serviceManager) RegisterProcessor(p Processor, msgType network.PacketTy
 }
 
 // TODO
-func (s *serviceManager) AvailableServices() []string {
-	panic("not implemented")
+func (s *serviceManager) AvailableServices() (ret []string) {
```
Comment (nikkolasg) [inline sda/service.go]: Need comment on that one.

File: sda/service.go [hunk 2]
```diff
@@ -272,19 +272,21 @@ func (s *serviceManager) RegisterProcessor(p Processor, msgType network.PacketTy
...
+	for id := range s.services {
+		ret = append(ret, ServiceFactory.Name(id))
+	}
+	return
 }
 
 // Service returns the Service implementation being registered to this name
 // TODO use serviceByString not implemented
```
Comment (nikkolasg) [inline sda/service.go]: TODO can be removed here.

**Comment by ineiti** [comment]
Crap - why does golint pass on those? It should've failed...

File: sda/service.go [hunk 3]
```diff
@@ -260,31 +260,33 @@ func (s *serviceManager) Process(data *network.Packet) {
...
+// AvailableServices returns a list of all services available to the serviceManager.
+// If no services are instantiated, it returns an empty list.
+func (s *serviceManager) AvailableServices() (ret []string) {
+	for id := range s.services {
+		ret = append(ret, ServiceFactory.Name(id))
+	}
+	return
 }
 
 // Service returns the Service implementation being registered to this name
```
Comment (nikkolasg) [inline sda/service.go]: `... or nil if no services have been found by this name`

**Comment by Liamsi** [comment]
LGTM

**Comment by nikkolasg** [comment]
After last comment, OK to merge.

**Comment by ineiti** [comment]
OK - done.

**Pull request merged**